website/source/docs/guides/consul-splitting.md

```diff
@@ -0,0 +1,461 @@
+proxy, and Envoy gets its configuration from Consul via gRPC.
+
+- Central service configuration is enabled - you will use configuration entries
+to specify that the API service protocol, and define your splitting ratios.
+
+The Consul configuration file at `consul_config/consul.hcl` contains the
+follwoing.
+
+```hcl
+data_dir = "/tmp/"
+log_level = "DEBUG"
+server = true
+
+bootstrap_expect = 1
+ui = true
+
+bind_addr = "0.0.0.0"
+client_addr = "0.0.0.0"
+
+connect {
+  enabled = true
+}
+
+ports {
+  grpc = 8502
+}
+
+enable_central_service_config = true
+```
+
+You can find the service definitions for this demo in the `service_config`
+folder.
+
+Once everything is up and running, you can view the health of the registered
+services by checking the Consul UI at `http://localhost:8500`. All services
+should be passing their health checks.
+
+![List of services in the Consul UI including Consul, and the web and API services with their proxies](/static/img/consul-splitting-services.png)
+
+Curl the Web endpoint to make sure that the whole application is running. The
+Web service will get a response from version 1 of the API service.
+
+```hcl
+$ curl localhost:9090
+Hello World
+###Upstream Data: localhost:9091###
+  Service V1%
+```
+
+Initially, you will want to deploy version 2 of the API service to production
+without sending any traffic to it, to make sure that it performs well in a new
+environment. Prevent traffic from flowing to version 2 when you register it, you
+will preemptively set up a  traffic split to send 100% of your traffic to
+version 1 of the API service, and 0% to the not-yet-deployed version 2.
+Splitting the traffic makes use of the new Layer 7 features built into Consul
+Service Mesh.
+
+To deploy version 2 of your API service, you will:
+
+1. Set up a traffic split to make sure v2 doesn’t receive any traffic at first.
+1. Start an instance of the v2 API service in your production environment.
+1. Register v2 so that Consul can send traffic to it.
+1. Slowly shift traffic to v2 and a way from v1 until the new version is
+handling all the traffic.
+
+## Configure Traffic Splitting
+
+Traffic Splitting uses configuration entries (introduced in Consul 1.5 and 1.6)
+to centrally configure the services and Envoy proxies. There are three
+configuration entries you need to create to enable traffic splitting:
+
+- Service Defaults for the API service to set the protocol to HTTP.
+- Service Splitter which defines the traffic split between the service subsets.
+- Service Resolver which defines which service instances are version 1 and 2.
+
+### Configuring Service Defaults
+
+Traffic splitting requires that the upstream application uses HTTP, because
+splitting happens on layer 7 (on a request by request basis). You will tell
+Consul that your upstream service uses HTTP by setting the protocol in a
+“service defaults” configuration entry for the API service. This configuration
```

Comment:
I lean toward using plain language when you're talking about a Consul item and the code snippet format when referring to the field, or a specific file or directory. In this case, I think the noun would be "configuration entry" and the adjective describing it is "service defaults". Because it's modifying a noun it should be hyphenated. Ex: "service-defaults configuration entry". Hope that makes sense, and your instinct that this is weird is correct. Fixing it now :P